pgrx/src/array/flat_array.rs

```diff
@@ -64,14 +64,10 @@ where
 #[non_exhaustive]
 #[derive(Debug)]
 pub enum ArrayAllocError {
-    TooManyBytes {
-        over: usize,
-    },
-    TooManyElems {
-        over: usize,
-    },
+    TooManyBytes,
+    TooManyElems,
     /// One or more dimensions are zero
-    ZeroLenDim(usize),
+    ZeroLenDim,
```

Comment:
While returning the information we had was cute, it wasn't viable in all cases without compromising the code significantly, and it was always at risk of being incomplete, so I stopped doing that.